mode_dark_light.py

```diff
@@ -1149,8 +1149,13 @@ def clickCompileApp():
     if not fileName or fileName == "   Not selected yet":
         return
 
-    os.system("mkdir kernel")
-    os.chdir("kernel")
+    script_dir = os.path.dirname(os.path.abspath(__file__))
+
+    kernel_dir = os.path.join(script_dir, "kernel")

```

Comment:
IIRC, this `kernel` folder should be created within the `build` folder, right? You are now creating it inside the `repo/` where the `launch_script.py` stays?

Reply:
oh ok! let me revise. 

the core problem of current code is that it uses the current path which triggers the script, so not stable enough.